guides/Makefile

```
@@ -24,3 +24,6 @@ linkchecker-tryer:
 	find build -type f -name index\*html | xargs linkchecker -r1 -f common/linkchecker.ini --check-extern | ./scripts/linkchecker-tryer
 
 ccutil: $(SUBDIRS)
+
+toc: html
+	F2L=upstream_filename_to_satellite_link.json ruby scripts/extract_links_toc build/**/index-*.html > build/toc.json
```

Since you're already doing this for Satellite only, you can limit it to just `index-satellite.html`. And also use the shebang
```suggestion
	F2L=upstream_filename_to_satellite_link.json ./scripts/extract_links_toc build/**/index-satellite.html > build/toc.json
```

Then for the mapping file you can map `File.dirname($filename)`. In the JSON you'll only have a mapping of directory names:
```json
{
  "Administering_Project": "administering_red_hat_satellite"
}
```

Then you probably want to change the JSON filename to mention `directory` instead of `filename`.

---

I prefer specifying the file names explicitly. Downstream the files and directories structure looks a bit different.

Once we unify the upstream and downstream - that would be an option.